app/events/applications/application_deleted_event.rb

```diff
@@ -7,9 +7,10 @@ def self.create(application)
     new(
       application: MissingModel::MissingApplication.new(id: application.id),
       metadata: {
-        provider_id: application.provider_account_id || application.tenant_id,
+        provider_id: application.provider_account.try(:id) || application.tenant_id, # TODO: I don't know why but the provider_account_id is not the same as provider_account.id
```

Comment:
Oh that is because of

https://github.com/3scale/porta/blob/master/app/models/contract.rb#L42
And

https://github.com/3scale/porta/blob/master/app/models/contract.rb#L126-L128

And the plan was created with another provider :/